CONTRIBUTING.md

```diff
@@ -1,6 +1,65 @@
-## Contributing
+# Contributing
 
-For now `master` is the main branch and code improvements are made in topic branches that get merged into it. Eventually there will be specific release branches, but for now `master` is the release branch.
+Planetary is an open-source project, as such, we openly welcome contributions of any sort: code improvement, bug fixes, translations, new features, bug reports...
+
+We encourage you to read this guide first or contact any of us.
+
+## Architecture
+
+**A good place to start: all of our important architectural decisions are being annotated inside the [Architecture folder](Architecture/) using [Architecture Decision Records](http://thinkrelevance.com/blog/2011/11/15/documenting-architecture-decisions).**
+
+Planetary is organized around [major components](https://developer.apple.com/documentation/swift_packages/organizing_your_code_with_local_packages). A quick look to the dependency graph:
+
+```mermaid
+graph TD
+  Planetary((Planetary))
+  Planetary-->Logger
+  Planetary-->Secrets
+  Planetary-->Analytics
+  Planetary-->CrashReporting
+  Secrets-->Logger
+  Analytics-->Logger
+  Analytics-->Secrets
+  CrashReporting-->Logger
+  CrashReporting-->Secrets
+```
+
+You can contribute by working on each of these packages, or in the Planetary app itself.
+
+### Creating a Swift Package
+
+In order to create a new package, follow these steps:
+
+1. In Xcode, select File > New > Package...
+2. Give it a name and be sure to select the Planetary workspace under `Add to` and `Group`
+3. Add the package in `Framework, Libraries, and Embedded Content` under the `Planetary` target (in the `General` tab) and under `Link Binary with Libraries` under the `UnitTests` target (in the `Build phases` tab)
+4. Be sure to make tests _execute in parallel_ and _in random order_ by turning on both options in Edit scheme > Test > Info
+5. Be sure to enable _code coverage_ when running tests by turning on that option in Edit scheme > Test > Options
+6. Add a new job in `.github/workflows/main.yml` just like the other packages
+
+You are ready to go. Please, use other local swift packages like `Analytics` or `Logger` as a model to develop the new package and **write tests**, we are aiming to a code coverage of more than 90% in our packages.
+
+### The Secrets.plist configuration file
+
+In order to configure third-party libraries, we need to add API keys. They are listed in the [Secrets.debug.plist](Resources/Secrets.debug.plist) file with empty values, the app still works without them. When making a Release build, you will need a similar file named Secrets.release.plist added in the same folder.
+
+Be sure to execute this in your local environment:
+
+```git update-index --skip-worktree Resources/Secrets.debug.plist```
+
+This prevent you from commiting real values in the repository.
+
+## Bug Reports
```

Comment:
Maybe it would make sense to move the Bug Reports, Translations, and Pull Request sections up a little higher in the file, because I think it will be relevant to more people than the Secrets or Swift Package Creation.